spring-cloud-gcp-logging/src/main/java/org/springframework/cloud/gcp/logging/TraceIdLoggingWebMvcInterceptor.java

```diff
@@ -35,22 +34,33 @@
  * trace ID metadata to log messages.
  *
  * @author Mike Eltsufin
+ * @author Chengyuan Zhao
  */
 public class TraceIdLoggingWebMvcInterceptor extends HandlerInterceptorAdapter {
 
-	private static final String X_CLOUD_TRACE = "x-cloud-trace-context";
+	private TraceIdExtractor traceIdExtractor;
 
-	private static final String X_B3_TRACE = "X-B3-TraceId";
+	public TraceIdLoggingWebMvcInterceptor(TraceIdExtractor extractor) {
+		Assert.notNull(extractor, "A valid trace id extractor is required.");
+		this.traceIdExtractor = extractor;
+	}
+
+	public TraceIdExtractor getTraceIdExtractor() {
+		return this.traceIdExtractor;
+	}
 
-	private static final List<String> TRACE_HEADERS = ImmutableList.of(X_CLOUD_TRACE,
-			X_B3_TRACE);
+	public void setTraceIdExtractor(
+			TraceIdExtractor traceIdExtractor) {
```

Comment:
I would stay away of the possible mutation at runtime. So, only ctor should be available.